cinema/tests/test_movie_api.py

```diff
@@ -128,0 +1,126 @@
+        for field in titanic:
+            self.assertEqual(movies.data[0][field], titanic[field])
+
+    def test_post_movies(self):
+        movies = self.client.post(
+            "/api/cinema/movies/",
+            {
+                "title": "Superman",
+                "description": "Superman description",
+                "duration": 123,
+                "actors": [1],
+                "genres": [1, 2],
+            },
+        )
+        db_movies = Movie.objects.all()
+        self.assertEqual(movies.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(db_movies.count(), 2)
+        self.assertEqual(db_movies.filter(title="Superman").count(), 1)
+
+    def test_post_invalid_movies(self):
+        movies = self.client.post(
+            "/api/cinema/movies/",
+            {
+                "title": "Superman",
+                "description": "Superman description",
+                "duration": 123,
+                "actors": [
+                    {
+                        "id": 3,
+                    }
+                ],
+            },
+        )
+        superman_movies = Movie.objects.filter(title="Superman")
+        self.assertEqual(movies.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(superman_movies.count(), 0)
+
+    def test_get_movie(self):
+        response = self.client.get("/api/cinema/movies/1/")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["title"], "Titanic")
+        self.assertEqual(response.data["description"], "Titanic description")
+        self.assertEqual(response.data["duration"], 123)
+        self.assertEqual(response.data["genres"][0]["name"], "Drama")
+        self.assertEqual(response.data["genres"][1]["name"], "Comedy")
+        self.assertEqual(response.data["actors"][0]["first_name"], "Kate")
+        self.assertEqual(response.data["actors"][0]["last_name"], "Winslet")
+        self.assertEqual(
+            response.data["actors"][0]["full_name"], "Kate Winslet"
+        )
+
+    def test_get_invalid_movie(self):
+        response = self.client.get("/api/cinema/movies/100/")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_put_movie(self):
+        self.client.put(
+            "/api/cinema/movies/1/",
+            {
+                "title": "Watchman",
+                "description": "Watchman description",
+                "duration": 321,
+                "genres": [1, 2],
+                "actors": [1],
+            },
+        )
+        db_movie = Movie.objects.get(id=1)
+        self.assertEqual(
+            [db_movie.title, db_movie.description],
+            [
+                "Watchman",
+                "Watchman description",
+            ],
+        )
+        self.assertEqual(db_movie.title, "Watchman")
+
+    def test_delete_movie(self):
+        response = self.client.delete(
+            "/api/cinema/movies/1/",

```

Comment:
Consider using Django's `reverse` function to generate the URL for the API endpoint. This will make the test more robust against changes in the URL patterns.